apps/server/src/modules/role/service/role.service.spec.ts

```diff
@@ -33,6 +33,12 @@ describe('RoleService', () => {
 			return id === testRoleEntity.id ? Promise.resolve(testRoleEntity) : Promise.reject();
 		});
 		roleRepo.findByNames.mockImplementation(async (names: RoleName[]): Promise<Role[]> => {
+			if (names.includes(RoleName.TEACHER) && names.includes(RoleName.ADMINISTRATOR)) {
```

Comment:
probably not your code, but I really dislike the usage of a mockfunction here. It would be absolutely sufficient to simply set up the function to return whatever needed per testcase or testsetup. In this position, this will become more and more complicated the more testcases are added.

Please refactor this.

Reply:
Refactored